src/status_im/contexts/wallet/send/utils.cljs

```diff
@@ -20,3 +23,36 @@
                        value1)))
              {}
              transaction-hashes))
+
+(defn network-amounts
+  [{:keys [route token-decimals native-token? to?]}]
+  (letfn [(merge-sum [a b]
+            (merge-with money/add a b))]
+    (reduce (fn [acc path]
+              (let [amount-hex   (if to? (:amount-in path) (:amount-out path))
+                    amount-units (native-module/hex-to-number
+                                  (utils.hex/normalize-hex amount-hex))
+                    amount       (money/with-precision
+                                  (if native-token?
+                                    (money/wei->ether amount-units)
+                                    (money/token->unit amount-units
+                                                       token-decimals))
+                                  6)
+                    chain-id     (if to? (get-in path [:to :chain-id]) (get-in path [:from :chain-id]))]
+                (merge-sum acc {chain-id amount})))
+            {}
+            route)))
```

Comment:
The uses of `merge-sum` is an interesting use of `merge-with`, It's really good to see we are starting to use a little more advanced Clojure functions :+1: .

But, in this case I think it's not needed, and IMO makes the code harder to understand, also computationally more expensive.

`merge-sum` is adding the keys to `acc` and if they already exist, will apply  `add` on the previous and the new value. Since we know the new key to add (`chain-id`), then we can instead  use an `update`:

```clojure
(update acc chain-id money/add amount)
```
This code will work even if the key doesn't exist in `acc` because  `money/add` takes `nil`s as `0`.

The code is more performant because we are not using `merge` to check all the existing keys in `acc`.